File: web/src/__generated/model/static/CurrentQueueDto.ts
import type {PlaybackStatus, PlaybackStrategy, StopStrategy} from '../enums/';
import type {CurrentQueueItemDto} from './';

export interface CurrentQueueDto {
    readonly items: ReadonlyArray<CurrentQueueItemDto>;
    readonly recordingIds: ReadonlyArray<number>;
    readonly currentIndex: number;
    readonly playbackStrategy: PlaybackStrategy;
    readonly stopStrategy: StopStrategy;
    readonly playbackStatus: PlaybackStatus;
    readonly positionMs: number;
    readonly serverTimeToExecuteMs: number;
    readonly version: number;
    readonly updatedAtMs: number;
}

File: server/src/main/kotlin/com/coooolfan/unirhy/sync/protocol/CurrentQueueDto.kt
package com.coooolfan.unirhy.sync.protocol

enum class PlaybackStrategy {
    SEQUENTIAL,
    SHUFFLE,
    RADIO,
}

enum class StopStrategy {
    TRACK,
    LIST,
}

data class CurrentQueueItemDto(
    val recordingId: Long,
    val title: String,
    val artistLabel: String,
    val coverUrl: String? = null,
    val durationMs: Long,
)

data class CurrentQueueDto(
    val items: List<CurrentQueueItemDto>,
    val recordingIds: List<Long>,
    val currentIndex: Int,
    val playbackStrategy: PlaybackStrategy,
    val stopStrategy: StopStrategy,
    val playbackStatus: PlaybackStatus,
    val positionMs: Long,
    val serverTimeToExecuteMs: Long,
    val version: Long,
    val updatedAtMs: Long,
)

data class QueueChangePayload(
    val queue: CurrentQueueDto,
)
